engine-tests/src/test/java/org/terasology/metatesting/EntitySystemTest.java

```diff
@@ -0,0 +1,292 @@
+    }
+
+    @Test
+    public void testComponentManagement() {
+        EntityRef entity = entityManager.create();
+        System.out.println("Starting testComponentManagement with entity: " + entity);
+
+        // 1. Add Component
+        TestComponent comp = new TestComponent();
+        comp.name = "Test Name";
+        entity.addComponent(comp);
+
+        assertTrue(entity.hasComponent(TestComponent.class));
+        assertEquals("Test Name", entity.getComponent(TestComponent.class).name);
+
+        // 2. Update Component
+        TestComponent comp2 = entity.getComponent(TestComponent.class);
+        comp2.name = "Updated Name";
+        entity.saveComponent(comp2);
+
+        assertEquals("Updated Name", entity.getComponent(TestComponent.class).name);
+
+        // 3. Remove Component
+        entity.removeComponent(TestComponent.class);
+        assertFalse(entity.hasComponent(TestComponent.class));
+    }
+
+    @Test
+    public void testSystemLifecycle() {
+        ComponentSystemManager systemManager = new ComponentSystemManager(context);
+        ComponentSystem mockSystem = mock(ComponentSystem.class);
+
+        systemManager.register(mockSystem);
+        System.out.println("System registered to systemManager: " + systemManager + " with mockSystem: " + mockSystem);
+
+        // Initialise should not be called yet
+        verify(mockSystem, never()).initialise();
+
+        systemManager.initialise();
+        verify(mockSystem).initialise();
+
+        systemManager.shutdown();
+        verify(mockSystem).shutdown();
+    }
+
+    @Test
+    public void testEventProcessing() {
+        // 1. Setup Event System and Entity
+        EntityRef entity = entityManager.create();
+        TestEvent event = new TestEvent();
+        System.out.println("Starting testEventProcessing with entity: " + entity + " and event: " + event);
+
+        // 2. Register Handlers via ComponentSystemManager (simulating real engine flow)
+        ComponentSystemManager systemManager = new ComponentSystemManager(context);
+
+        TestEventHandler handlerNormal = new TestEventHandler();
+        TestEventHandlerHighPriority handlerHigh = new TestEventHandlerHighPriority();
+
+        systemManager.register(handlerNormal);
+        systemManager.register(handlerHigh);
+        systemManager.initialise();
+        System.out.println("Handlers registered and manager initialised to systemManager: " + systemManager);
+
+        // 3. Fire Event
+        System.out.println("Sending event: " + event);
+        entity.send(event);
+
+        // 4. Verify Priority (High should run first)
+        assertTrue(handlerHigh.received, "High priority handler should have received event");
+        assertTrue(handlerNormal.received, "Normal priority handler should have received event");
+
+        // In a real scenario we'd verify order, but basic receipt is a good start.
+        // Since we didn't consume it, both should get it.
```

Comment:
This comment implies that the test is incomplete? I would either remove it or find a way to explicitly test the order of event handling.

Reply:
Yeah I noticed that too - pretty sure I had been discussing some advanced tests / going deeper with entities, but at the moment was trying to push forward to troubleshooting the multiplayer issue. Usually easy to point somewhere specific and essentially go "expand on this further" later :-)